pkg/scheduler/generic_scheduler_test.go

```diff
@@ -1068,11 +1069,9 @@ func TestFindFitAllError(t *testing.T) {
 		t.Errorf("unexpected error: %v", err)
 	}
 
-	// TODO(#94696): use cmp.Diff() to compare `diagnosis`.
-	if len(diagnosis.NodeToStatusMap) != len(nodes) {
-		t.Errorf("unexpected failed status map: %v", diagnosis.NodeToStatusMap)
+	if diff := cmp.Diff(len(diagnosis.NodeToStatusMap), len(nodes)); diff != "" {
```

Comment:
The intention was to use a single comparison on `Diagnosis` object, to replace the current logic that compares individual fields. Something like:

```go
	want := framework.Diagnosis{
		NodeToStatusMap: framework.NodeToStatusMap{
			"1": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("MatchFilter"),
			"2": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("MatchFilter"),
			"3": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("MatchFilter"),
		},
		UnschedulablePlugins: sets.NewString("MatchFilter"),
	}
	if diff := cmp.Diff(diagnosis, want); diff != "" {
		t.Errorf("Unexpected diagnosis: (-want, +got): %s", diff)
	}
```

Reply:
The relevant code has been modified,thx